traitsui/testing/tester/_ui_tester_registry/_traitsui_ui.py

```diff
@@ -9,7 +9,7 @@
 #  Thanks for using Enthought open source!
 #
 
-from traitsui.testing.api import TargetById, TargetByName
+from traitsui.testing.tester.locator import TargetById, TargetByName
```

Comment:
I think this makes more sense for the internal non-test code NOT to depend on `api` . So this change is good.

The audience of `api` is the user of the library. It makes discovery of public API easier and it encapsulates internal structure from the users. In order to do so, it brings together all public API the library has to offer, i.e. it _depends_ on the library internal code.  For the library internal code to depend on its own `api` again, cycle dependency is created and cycle imports would be inevitable.

It is fine for test code to import from `api` though (e.g. `test_button_editor`), because test is a component isolated from the software being tested, so it is allowed to depend on `api` like the user would do. In fact, it is better for tests to import from `api` as it exercises the user's context better.

I think previously this has worked based on chance 😓 . The import that has the most dependency is `ui_tester` (it is the top-level thing). It was imported late in `api` by virtue of alphabetical order. The cycle dependency is there but cycle imports were not triggered because Python import caches imported modules.

Reply:
I removed a few more API imports on this PR that were being used by internal non-test code.  All of the editor implementations currently import from API - I think I will replace these in a separate PR.